scripts/skills/perks/perk_rf_iron_sights.nut

```diff
@@ -15,16 +15,30 @@ this.perk_rf_iron_sights <- ::inherit("scripts/skills/skill", {
 		this.m.IsHidden = false;
 	}
 
+	function isEnabled()
+	{
+		if (this.getContainer().getActor().isDisarmed())
+			return false;
+
+		local weapon = this.getContainer().getActor().getMainhandItem();
+		return weapon != null && weapon.isItemType(::Const.Items.ItemType.RangedWeapon) && (weapon.isWeaponType(::Const.Items.WeaponType.Crossbow) || weapon.isWeaponType(::Const.Items.WeaponType.Firearm));
+	}
+
 	function onUpdate( _properties )
 	{
-		if (this.getContainer().getActor().isDisarmed()) return;
+		if (this.isEnabled())
+			_properties.HitChance[::Const.BodyPart.Head] += this.m.Bonus;
+	}
+
+	function onTargetHit( _skill, _targetEntity, _bodyPart, _damageInflictedHitpoints, _damageInflictedArmor )
+	{
+		if (!_targetEntity.isAlive() || _bodyPart != ::Const.BodyPart.Head || _targetEntity.getMoraleState() == ::Const.MoraleState.Ignore || !_skill.isRanged() || !_skill.isAttack() || !this.isEnabled())
```

Comment:
How does the morale state `::Const.MoraleState.Ignore` make an enemy immune to being shellshocked?

I think this condition only existed in vanilla because they do a negative morale check alongside applying shellshocked.
And I even consider it a vanilla bug, that mortars can't apply shellshocked on enemies which ignore Morale.

Reply:
I think vanilla did this to prevent undead units from getting shellshocked which kinda makes sense because they don't suffer from mental and physical trauma like that. I just added this condition to be in line with vanilla conditions for applying shellshocked. But we can discuss it and can replace it with a check for the entity being undead instead of checking the morale state.

Reply:
That's a good idea, I'll add it to the tooltip.